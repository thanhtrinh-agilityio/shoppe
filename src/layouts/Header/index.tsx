import { useMemo } from 'react'
import Link from 'next/link'
import Image, { ImageLoader } from 'next/image'
import { Box, Flex, Heading, Text } from '@chakra-ui/react'

// Components
import Icon from '@components/Icon'

// Services
import myImageLoader from '@services/imageLoader'

// Hooks
import { useCartContext } from '@hooks/useCartContext'

// Types
import { IProductDetail } from '@self-types/index'

const Header = () => {
  const { listCart } = useCartContext()

  const quantity = useMemo(
    () =>
      listCart.reduce((acc: number, obj: IProductDetail) => {
        return acc + obj.quantity
      }, 0),
    [listCart],
  )

  return (
    <Flex
      as="header"
      borderBottom="1px"
      borderBottomColor="lightSilver"
      justifyContent="space-between"
      alignItems="center"
      paddingBottom="5px"
      maxWidth="1248px"
      margin="0 auto"
      pt="67px"
    >
      <Heading as="h1">
        <Link href="/">
          <Image
            width={125}
            height={40}
            src="/images/logo.png"
            alt="Logo Shoppe"
            loader={myImageLoader as ImageLoader}
          />
        </Link>
      </Heading>
      <Box position="relative">
        <Icon
          width="21px"
          height="20px"
          marginRight="60px"
          srcIcon="/images/shoppingCart.svg"
          alt="Shopping Cart Icon"
        />
        <Flex
          bottom="14px"
          left="15px"
          w="20px"
          height="20px"
          position="absolute"
          background="warning"
          borderRadius="4px"
          alignItems="center"
          justifyContent="center"
        >
          <Text size="small" color="gray">
            {quantity}
          </Text>
        </Flex>
      </Box>
    </Flex>
  )
}

export default Header
