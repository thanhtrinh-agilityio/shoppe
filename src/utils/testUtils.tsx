import { ReactNode } from 'react'
import { render } from '@testing-library/react'
import { ChakraProvider } from '@chakra-ui/react'

// Themes
import { customTheme } from '@themes/index'

// Contexts
import { CartProvider } from '@contexts/CartProvider'

export const customRender = (component: ReactNode) => {
  return render(
    <ChakraProvider theme={customTheme}>
      <CartProvider>{component}</CartProvider>
    </ChakraProvider>,
  )
}

export * from '@testing-library/react'
export * from '@testing-library/jest-dom'
export { customRender as render }
