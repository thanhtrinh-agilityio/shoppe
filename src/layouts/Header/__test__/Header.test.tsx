// Utils
import { render } from '@utils/testUtils'

// Components
import Header from '@layouts/Header'

describe('Header render', () => {
  it('Should Header match Snapshot', () => {
    const { container } = render(<Header />)

    expect(container).toMatchSnapshot()
  })
})
