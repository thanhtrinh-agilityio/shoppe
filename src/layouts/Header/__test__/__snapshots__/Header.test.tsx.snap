// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header render Should Header match Snapshot 1`] = `
<div>
  <header
    class="css-n5jh5q"
  >
    <h1
      class="chakra-heading css-1dklj6k"
    >
      <img
        alt="Logo Shoppe"
        data-nimg="1"
        decoding="async"
        height="40"
        loading="lazy"
        src="/images/logo.png?w=256&q=75"
        srcset="/images/logo.png?w=128&q=75 1x, /images/logo.png?w=256&q=75 2x"
        style="color: transparent;"
        width="125"
      />
    </h1>
    <div
      class="css-79elbk"
    >
      <div
        class="css-c3i4bm"
      >
        <img
          alt="Shopping Cart Icon"
          data-nimg="fill"
          data-testid="image"
          decoding="async"
          loading="lazy"
          sizes="100vw"
          src="/images/shoppingCart.svg?w=3840&q=75"
          srcset="/images/shoppingCart.svg?w=640&q=75 640w, /images/shoppingCart.svg?w=750&q=75 750w, /images/shoppingCart.svg?w=828&q=75 828w, /images/shoppingCart.svg?w=1080&q=75 1080w, /images/shoppingCart.svg?w=1200&q=75 1200w, /images/shoppingCart.svg?w=1920&q=75 1920w, /images/shoppingCart.svg?w=2048&q=75 2048w, /images/shoppingCart.svg?w=3840&q=75 3840w"
          style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
        />
      </div>
      <div
        class="css-2jnr0l"
      >
        <p
          class="chakra-text css-1xb8dar"
        >
          0
        </p>
      </div>
    </div>
  </header>
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;
